mantidproject/mantid
# Engg GUI: change of units for fitting tab

**Description of work.**

By converting units of single peak fitting plot on `Fitting` Tab, this PR enables you to rerun the fitting process after selecting peaks from the interface. A change to the algorithm flow was required in order to clone every single workspace to the _input file_. `(focused bank workspace)`. The fitting plot can now be viewed in `dSpacing` instead `ToF`. 
- [x] _Will require to fix some conflicts once  #15823 is merged_

**To test:**



Files can be found at: `\\olympic\Babylon5\Public\Shahroz\`
Go to: `Interfaces/Diffraction/Engineering Diffraction/Fitting Tab`
Select a focused file to Fit : `ENGINX_241391_focused_texture_bank_3.nxs`
Add peaks: `3.1243,2.7057,1.9132`
Click `Fit`

Once fitted:
 Click `Select Peak` button
 This should enable the `peak picking tool`
 Select peak by holding `shift` key and `left click` on graph
 Do not select a peak that has already been fitted to avoid confusion

Once centre of peak selected:
 Click `Add Peak` button
  Which should add the selected peak to `Peaks` text-field 
  **Note the value of the peak**

Click `Fit` again..
Upon completion:
(`zoom-in` by left click and drag on plot)
- [x] Check that the selected peak has been Fitted by zooming-in 
- [x] Open `engggui_fitting_fitpeaks_params` table workspace and find a row of data for the **noted value of the peak**
- [x] Check that `engggui_fitting_single_peaks` is a clone of `engggui_fitting_focused_ws`

_Note: Saving the table workspace not included in this PR, as it required more feedback from instrument scientists_

Fixes #15766.
[Release Notes](https://github.com/mantidproject/mantid/blob/e3084ffc5636d71a5934c6e876f00c1a2c3d091f/docs/source/release/v3.7.0/diffraction.rst#id7)

  

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
